cli/runtime/config/config_factory.go

```diff
@@ -8,22 +8,194 @@ import (
 	"os"
 
 	"github.com/pkg/errors"
+
 	"gopkg.in/yaml.v3"
+
+	"github.com/vmware-tanzu/tanzu-framework/cli/runtime/config/nodeutils"
 )
 
-type CfgOptions struct {
-	CfgPath string // file path to the config file
+// getMultiConfig retrieves combined config.yaml and config-ng.yaml
+func getMultiConfig() (*yaml.Node, error) {
+	node1, err := getClientConfig()
+	if err != nil {
+		return node1, err
+	}
+
+	node2, err := getClientConfigNextGenNode()
+	if err != nil {
+		return node2, err
+	}
+
+	return makeMultiFileCfg(node1, node2)
+}
+
+// getMultiConfigNoLock retrieves combined config.yaml and config-ng.yaml
+func getMultiConfigNoLock() (*yaml.Node, error) {
+	cfgNode, err := getClientConfigNoLock()
+	if err != nil {
+		return cfgNode, err
+	}
+
+	cfgNextGenNode, err := getClientConfigNextGenNodeNoLock()
+	if err != nil {
+		return cfgNextGenNode, err
+	}
+
+	return makeMultiFileCfg(cfgNode, cfgNextGenNode)
 }
 
-type CfgOpts func(config *CfgOptions)
+// Construct multi file node from config.yaml and config-ng.yaml nodes
+func makeMultiFileCfg(node1, node2 *yaml.Node) (*yaml.Node, error) {
+	// config items that goes to config-ng.yaml
+	nextGenCfgItems := []string{
+		KeyContexts,
+		KeyCurrentContext,
+	}
+
+	cfgNode := node1.Content[0]
+
+	for _, nextGenItem := range nextGenCfgItems {
+		// Find the next gen item node from the config.yaml
+		nextGenItemNodeIndex := nodeutils.GetNodeIndex(cfgNode.Content, nextGenItem)
+		if nextGenItemNodeIndex == -1 {
+			continue
+		}
+		// Delete the next gen item node from the config.yaml
+		cfgNode.Content = append(cfgNode.Content[:nextGenItemNodeIndex-1], cfgNode.Content[nextGenItemNodeIndex:]...)
+		cfgNode.Content = append(cfgNode.Content[:nextGenItemNodeIndex-1], cfgNode.Content[nextGenItemNodeIndex:]...)
+	}
+
+	// Construct Multi Bytes data
+	var multiBytes []byte
+	var multiNode yaml.Node
+
+	// construct node1 bytes and append to multiBytes
+	if node1.Content[0].Content != nil && len(node1.Content[0].Content) > 0 {
+		node1Bytes, err := yaml.Marshal(node1)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(node1Bytes) != 0 {
+			multiBytes = append(multiBytes, node1Bytes...)
+		}
+	}
+
+	// construct node 2 bytes and append to multiBytes
+	if node2.Content[0].Content != nil && len(node2.Content[0].Content) > 0 {
+		node2Bytes, err := yaml.Marshal(node2)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(node2Bytes) != 0 {
+			multiBytes = append(multiBytes, node2Bytes...)
+		}
+	}
```

Comment:
better have a helper function to avoid code duplication

Reply:
refactored